File: 0467. Unique Substrings in Wraparound String/Solution.py
from collections import defaultdict


class Solution:
    def findSubstringInWraproundString(self, p: str) -> int:
        cnt = defaultdict(int)
        span = 0
        for i, c in enumerate(p):
            if i and (ord(c) - ord(p[i - 1]) + 26) % 26 == 1:
                span += 1
            else:
                span = 1
            cnt[c] = max(cnt[c], span)
        return sum(cnt.values())

File: 0467. Unique Substrings in Wraparound String/Solution.c
#include <math.h>

int findSubstringInWraproundString(char* p)
{
    int cnt[26] = {0}, len = 0;
    for (int i = 0; p[i]; ++i) {
        if (i && (p[i] - p[i - 1] + 26) % 26 == 1)
            ++len;
        else
            len = 1;
        cnt[p[i] - 'a'] = fmax(cnt[p[i] - 'a'], len);
    }
    int res = 0;
    for (int i = 0; i < 26; ++i)
        res += cnt[i];
    return res;
}
